consensus/babe/runtime.go

```diff
@@ -21,13 +21,22 @@ import (
 )
 
 // gets the configuration data for Babe from the runtime
-func (b *Session) configurationFromRuntime() (*BabeConfiguration, error) {
+func (b *Session) configurationFromRuntime() error {
 	ret, err := b.rt.Exec("BabeApi_configuration", 1, []byte{})
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	bc := new(BabeConfiguration)
+	bc.GenesisAuthorities = []AuthorityData{}
 	_, err = scale.Decode(ret, bc)
-	return bc, err
+
+	if err != nil {
+		return err
+	}
+
+	// Directly set the babe session's config
+	b.config = bc
+
+	return err
```

Comment:
since we already have the babe session as the receiver in ` (b *Session)`, it's a bit cleaner to just set it directly instead of returning it and setting it wherever this function is called. also in terms of functionality, we will only need to call this function once, when we set up the session for the first time

Reply:
Sweet, thanks for explaining!